File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/login/PortalsPagerFragment.java
package app.editors.manager.ui.fragments.login;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.ArrayList;
import java.util.List;

import app.editors.manager.R;
import app.editors.manager.ui.activities.login.PortalsActivity;
import app.editors.manager.ui.fragments.base.BaseAppFragment;
import app.editors.manager.ui.views.pager.PagingViewPager;
import app.editors.manager.ui.views.pager.ViewPagerAdapter;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

public class PortalsPagerFragment extends BaseAppFragment {

    public static final String TAG = PortalsPagerFragment.class.getSimpleName();

    @BindView(R.id.on_boarding_view_pager)
    protected PagingViewPager mViewpager;

    private PortalsActivity mPortalsActivity;
    private Unbinder mUnbinder;

    public static PortalsPagerFragment newInstance() {
        return new PortalsPagerFragment();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            mPortalsActivity = (PortalsActivity) context;
        } catch (ClassCastException e) {
            throw new RuntimeException(PortalsPagerFragment.class.getSimpleName() + " - must implement - " +
                    PortalsActivity.class.getSimpleName());
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        final View view = inflater.inflate(R.layout.fragment_login_pager, container, false);
        mUnbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init(savedInstanceState);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mUnbinder.unbind();
    }

    private void init(final Bundle savedInstanceState) {
        mPortalsActivity.getTabLayout().setupWithViewPager(mViewpager, true);
        mViewpager.setAdapter(new ViewPagerAdapter(getChildFragmentManager(), getFragments()));
    }

    private List<ViewPagerAdapter.Container> getFragments() {
        final List<ViewPagerAdapter.Container> pairs = new ArrayList<>();
        pairs.add(new ViewPagerAdapter.Container(EnterprisePortalFragment.newInstance(),
                getString(R.string.login_pager_enterprise)));
        pairs.add(new ViewPagerAdapter.Container(PersonalPortalFragment.newInstance(),
                getString(R.string.login_pager_personal)));
        return pairs;
    }

}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/login/PortalsPagerFragment.kt
package app.editors.manager.ui.fragments.login

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import app.editors.manager.BuildConfig
import app.editors.manager.R
import app.editors.manager.ui.activities.login.PortalsActivity
import app.editors.manager.ui.fragments.base.BaseAppFragment
import app.editors.manager.ui.views.pager.PagingViewPager
import app.editors.manager.ui.views.pager.ViewPagerAdapter

class PortalsPagerFragment : BaseAppFragment() {

    companion object {
        val TAG: String = PortalsPagerFragment::class.java.simpleName

        fun newInstance(): PortalsPagerFragment {
            return PortalsPagerFragment()
        }
    }


    private var viewpager: PagingViewPager? = null
    private var portalsActivity: PortalsActivity? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        portalsActivity = try {
            context as PortalsActivity
        } catch (e: ClassCastException) {
            throw RuntimeException(
                PortalsPagerFragment::class.java.simpleName + " - must implement - " +
                        PortalsActivity::class.java.simpleName
            )
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_login_pager, container, false)?.apply {
            viewpager = findViewById(R.id.on_boarding_view_pager)
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }


    private fun init() {
        portalsActivity?.tabLayout?.setupWithViewPager(viewpager, true)
        viewpager?.adapter = ViewPagerAdapter(childFragmentManager, fragments)
    }

    private val fragments: List<ViewPagerAdapter.Container> by lazy {
        val list = arrayListOf<ViewPagerAdapter.Container>()
        list.add(ViewPagerAdapter.Container(
            EnterprisePortalFragment.newInstance(),
            getString(R.string.login_pager_enterprise)
        ))
        if (BuildConfig.APPLICATION_ID == "com.onlyoffice.documents") {
            list.add(ViewPagerAdapter.Container(
                PersonalPortalFragment.newInstance(),
                getString(R.string.login_pager_personal)
            ))
        }
        return@lazy list
    }

}